src/components/PostDetails.tsx

```diff
@@ -2,106 +1,123 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { client, deleteComment } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [isWritingComment, setIsWritingComment] = useState(false);
+
+  const loadComments = async (postId: number) => {
+    setIsWritingComment(false);
+    setIsLoading(true);
+    setError('');
+
+    try {
+      const fetchedComments = await client.get<Comment[]>(
+        `/comments?postId=${postId}`,
+      );
+
+      setComments(fetchedComments);
+    } catch (err) {
+      setError('Something went wrong');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadComments(post.id);
+  }, [post]);
+
+  const removeComment = (id: number) => {
+    const previousComments = [...comments];
+
+    setComments(curr => curr.filter(comment => comment.id !== id));
+
+    deleteComment(id).catch(() => {
+      setComments(previousComments);
+      setError('Failed to delete the comment');
+    });
```

Comment:
Consider handling errors more gracefully when deleting a comment. Currently, it only resets the comments to the previous state and sets an error message. Providing more detailed feedback to the user could improve the experience.